src/TapeChart.js

```diff
@@ -0,0 +1,48 @@
+class TapeChart {
+  constructor(users, rooms, bookings) {
+    this.users = users;
+    this.rooms = rooms;
+    this.bookings = bookings;
+  }
+
+getAvailableRoomsByDate(date) {
+    let availableByDate = this.rooms.reduce((acc, room) => {
+      let bookingsByRoom = this.bookings.filter(booking =>
+        booking.roomNumber === room.number);
```

Comment:
Im not saying someone accidentally added a string to the bookings database, but if they did it would be best to wrap the booking.roomNumber in a parseInt()